### PR TITLE
test(api): test downtime on services

### PR DESCRIPTION
**Pull request by loiclau**
service downtime api test

### CODE REVIEW

File: src/behat/Api/Context/ApiContext.php [hunk 1]
```diff
@@ -504,4 +504,97 @@ function() use ($host, $service, &$hostId, &$serviceId) {
 
         return [$hostId, $serviceId];
     }
+
+    /**
+     * Send a downtime to a service
+     *
+     * @When I send a downtime on my service :service from host :host
+     */
+    public function iSendDonwtimeOnServiceFromHost(string $service, string $host)
```
Comment (jeremyjaouen) [inline src/behat/Api/Context/ApiContext.php:513]:
```suggestion
    public function iSendDowntimeOnServiceFromHost(string $service, string $host)
```

File: src/behat/Api/Context/ApiContext.php [hunk 2]
```diff
@@ -504,4 +504,97 @@ function() use ($host, $service, &$hostId, &$serviceId) {
...
+            $data
+        );
+    }
+
+    /**
+     * List downtime for a service
+     *
+     * @When I list downtime of my service :servcie from host :host
+     */
+    public function iListDonwtimeOfServiceFromHost(string $service, string $host)
```
Comment (jeremyjaouen) [inline src/behat/Api/Context/ApiContext.php:544]:
```suggestion
    public function iListDowntimeOfServiceFromHost(string $service, string $host)
```

File: src/behat/Api/Context/ApiContext.php [hunk 3]
```diff
@@ -504,4 +504,97 @@ function() use ($host, $service, &$hostId, &$serviceId) {
...
+        $response = json_decode($response->getContent(), true);
+        return $response['result'][0]['id'];
+    }
+
+    /**
+     * Cancel downtime for a service
+     *
+     * @When I cancel downtime of my service :service from host :host
+     */
+    public function iCancelDonwtimeOfServiceFromHost(string $service, string $host)
```
Comment (jeremyjaouen) [inline src/behat/Api/Context/ApiContext.php:570]:
```suggestion
    public function iCancelDowntimeOfServiceFromHost(string $service, string $host)
```

File: src/behat/Api/Context/ApiContext.php [hunk 4]
```diff
@@ -504,4 +504,97 @@ function() use ($host, $service, &$hostId, &$serviceId) {
...
+    {
+        $downtimeId = $this->iListDonwtimeOfServiceFromHost($service, $host);
```
Comment (jeremyjaouen) [inline src/behat/Api/Context/ApiContext.php:572]:
```suggestion
        $downtimeId = $this->iListDowntimeOfServiceFromHost($service, $host);
```

File: src/behat/Api/Context/ApiContext.php [hunk 5]
```diff
@@ -504,4 +504,97 @@ function() use ($host, $service, &$hostId, &$serviceId) {
...
+            '/beta/monitoring/downtimes/' . $downtimeId
+        );
+    }
+
+    /**
+     * check if a downtime for a service is canceled
+     *
+     * @When the downtime of my service :service from host :host is cancel
+     */
+    public function theDonwtimeOfServiceFromHostIsCancel(string $service, string $host)
```
Comment (jeremyjaouen) [inline src/behat/Api/Context/ApiContext.php:585]:
```suggestion
    public function theDowntimeOfServiceFromHostIsCancel(string $service, string $host)
```

File: src/behat/Api/Context/ApiContext.php [hunk 6]
```diff
@@ -504,4 +504,97 @@ function() use ($host, $service, &$hostId, &$serviceId) {
...
+     */
+    public function iListDonwtimeOfServiceFromHost(string $service, string $host)
+    {
+        [$hostId, $serviceId] = $this->iWaitUntilServiceIsMonitored($service, $host);
+
+        $response = $this->iSendARequestTo(
+            'GET',
+            '/beta/monitoring/hosts/' . $hostId . '/services/' . $serviceId . '/downtimes'
+        );
+        $this->container->execute("php /usr/share/centreon/cron/downtimeManager.php", 'web');
```
Comment (kduret) [inline src/behat/Api/Context/ApiContext.php:552]: Why do you run downtimeManager ? You add a realtime downtime using api, isn't it ?

File: src/behat/Api/Context/ApiContext.php [hunk 7]
```diff
@@ -504,4 +504,97 @@ function() use ($host, $service, &$hostId, &$serviceId) {
...
+        $this->spin(
+            function () {
+                $this->theJsonNodeShouldHaveElements('result', 1);
+                return true;
+            },
+            'the host ' . $service . ' seems not monitored',
```
Comment (kduret) [inline src/behat/Api/Context/ApiContext.php:558]:
```suggestion
            'the service ' . $service . ' from host ' . $host . ' seems not monitored',
```

File: src/behat/Api/Context/ApiContext.php [hunk 8]
```diff
@@ -504,4 +504,97 @@ function() use ($host, $service, &$hostId, &$serviceId) {
...
+
+        $response = $this->iSendARequestTo(
+            'GET',
+            '/beta/monitoring/hosts/' . $hostId . '/services/' . $serviceId . '/downtimes'
+        );
+        $this->theJsonNodeShouldHaveElements('result', 1);
+        $response = json_decode($response->getContent(), true);
+
+        if (!$response['result'][0]['is_cancelled']) {
+            throw new \Exception('the service ' . $service . ' is not cancel');
```
Comment (kduret) [inline src/behat/Api/Context/ApiContext.php:597]:
```suggestion
            throw new \Exception('the downtime of service ' . $service . ' from host ' . $host . ' is not cancel');
```

**Comment by kduret** [comment]
I'm not sure downtime stuff should be is this repository, since it will not be necessarily used by modules

File: src/behat/Api/Context/ApiContext.php [hunk 9]
```diff
@@ -504,4 +504,97 @@ function() use ($host, $service, &$hostId, &$serviceId) {
 
         return [$hostId, $serviceId];
     }
+
+    /**
+     * Send a downtime to a service
+     *
+     * @When I send a downtime on my service :service from host :host
+     */
+    public function iSendDowntimeOnServiceFromHost(string $service, string $host)
```
Comment (sc979) [inline src/behat/Api/Context/ApiContext.php:513]: these methods shouldn't be added to the apiContext, but in the centreon downtime feature context

**Pull request closed without merging**